### PR TITLE
Use dagre and graphlib for layout, pixi.js for render

### PR DESCRIPTION
**Pull request by mzgoddard**
The graph is created with nodes and edges added in the devtools page.
Pass the graph to the devtools panel, which sets node sizes, and passes
to a web worker. On the web worker dagre lays out the graph and sends it
back to the panel. When the panel receives it, it updates the positions
of nodes and updates edge lines.

### CODE REVIEW

File: src/panel/math.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Clamp a value between two extremes.
+ * @param {number} value
+ * @param {number} min
+ * @param {number} max
+ * @return {number}
+ */
+export function clamp(value, min, max) {
+  invariant(min <= max, 'clamp(_, min, max): min is less than max');
```
Comment (hoch) [inline src/panel/math.js:11]: What does `invariant` do?

**Review by hoch — APPROVED** [review]
Looks great, but does this PR include a working visualizer? Or is it still mostly infrastructure and utility?

**Comment by mzgoddard** [comment]
> Looks great, but does this PR include a working visualizer? Or is it still mostly infrastructure and utility?

It has a working visualizer. It is mostly what I demoed last with a number of fixes.

File: package.json [hunk 2]
```diff
@@ -21,8 +21,11 @@
   "devDependencies": {
     "@babel/core": "^7.14.6",
     "@babel/plugin-transform-modules-commonjs": "^7.14.5",
+    "@pixi/events": "^6.1.0-rc",
     "@pixi/jsdoc-template": "^2.6.0",
+    "@pixi/unsafe-eval": "^6.0.4",
```
[the remark this answers is not in the document]
Comment (mzgoddard) [reply package.json:26]: The title of the package is a bit misleading. PIXI.js normally uses `new Function` to improve performance. This package disables use of `new Function` and uses a slower method.

[The package](https://github.com/pixijs/pixijs/tree/dev/packages/unsafe-eval) describes it this way:

> Adds support for environments that disallow support of new Function, such as WeChat.

File: src/panel/math.js (hunk 1 above)
Comment (hoch) [reply src/panel/math.js:11]: Could you explain this?

**Pull request merged**